crates/server/src/server/server_future.rs

```diff
@@ -64,8 +64,9 @@ impl<T: RequestHandler> ServerFuture<T> {
         let handler = self.handler.clone();
 
         // this spawns a ForEach future which handles all the requests into a Handler.
-        let task = tokio::spawn({
+        self.join_set.spawn({
```

Comment:
I'm a little worried that we have a pattern here that we need to ensure is followed, but isn't necessarily what people expect, i.e. use `join_set::spawn` and not `tokio::spawn`.

I think we can achieve this with clippy, in `clippy.toml`:

```
disallowed-methods = [
    # fully qualified function/method name:
    "tokio::spawn",
]
```

Reply:
Will that prevent consumers from spawning other unrelated detached tasks within a binary they're writing?

Reply:
I think it should only impact this project, not any that depends on it.

Reply:
Ok. If it’s too noisy, this is fine. I’ll merge as is and we can revisit later. I’ll play with some other options.

Reply:
And thank you for looking into that option. I appreciate the time and energy you put into it.